packages/travelmakers-design-core/src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,305 @@
+      ) {
+        return true;
+      }
+      return false;
+    };
+
+    const isMinNightDays = (day: DateCellDay) => {
+      if (checked.from) {
+        const resultDay = differenceInDays(day.date, checked.from.date);
+        if (resultDay > minNight) return true;
+        return false;
+      }
+      return false;
+    };
+
+    /**
+     * ANCHOR: From Date, To Date 날짜 사이에 대해서 체크
+     * @param day
+     * @returns
+     */
+    const isBetweenFromAndToDays = (day: DateCellDay) => {
+      if (
+        checked.from &&
+        checked.to &&
+        differenceInDays(day.date, checked.from.date) > 0 &&
+        differenceInDays(checked.to.date, day.date) > 0
+      ) {
+        return true;
+      }
+      return false;
+    };
+
+    /**
+     * ANCHOR: 선택 불가능한 날짜(disabledDays) 사이에 대해서 체크
+     * @param day
+     * @returns
+     */
+    const isDisabledDay = (day: DateCellDay) => {
+      let isCloset = true;
+      for (const disabledDay of disabledDays) {
+        if (isEqual(disabledDay, day.date)) {
+          isCloset = true;
+          break;
+        }
+      }
+
+      for (const selectableDate of selectableDates) {
+        if (isEqual(selectableDate, day.date)) {
+          isCloset = false;
+          break;
+        }
+      }
+      return isCloset;
+    };
+
+    const onClear = () => {
+      setChecked({ to: null, from: null });
+    };
+
+    const onClick = (day: DateCellDay) => {
+      if (isDisabledDay(day)) {
+        toast({
+          text: notAllowedMessage,
+        });
+      } else if (type === "tour") {
+        setChecked((prev) => {
+          return { ...prev, from: day, to: day };
+        });
+      } else if (!checked.from) {
+        setChecked((prev) => {
+          return { ...prev, from: day };
+        });
+        setEnabledDays(addDays(day.date, maxNight));
+      } else if (
+        isBetweenNotSelectedDays(day) &&
+        (isDisabledDay(day) || !isMinNightDays(day))
+      ) {
+        toast({
+          text: notAllowedMessage,
+        });
+      } else if (isBetweenNotSelectedDays(day)) {
+        setChecked((prev) => {
+          return { ...prev, to: day };
+        });
+        setEnabledDays(addDays(day.date, maxNight));
+      } else {
+        setChecked({ to: null, from: null });
+      }
+    };
+
+    const onType = (day: DateCellDay): DateCellType => {
+      if (isDisabledDay(day)) {
+        if (checked.from && isEqual(checked.to?.date, day.date)) {
+          return "disabled-to-between";
+        } else if (
+          checked.from &&
+          differenceInDays(day.date, checked.from.date) > 0 &&
+          differenceInDays(enabledDays, day.date) >= 0 &&
+          (!checked.to || differenceInDays(checked.to?.date, day.date) >= 0)
+        ) {
+          if (differenceInDays(enabledDays, day.date) > 0) {
+            return "disabled-between";
+          } else {
+            return "disabled-to-between";
+          }
```

Comment:
P3; 이부분도 중첩을 조금 줄이기 위해 삼항 연산자로 하는건 어떨까요?

```suggestion
          differenceInDays(enabledDays, day.date) > 0 ? "disabled-between" : "disabled-to-between"
```

Reply:
✅